include/mata/mintermization.hh

```diff
@@ -0,0 +1,143 @@
+/*
+ * mintermization.hh -- Mintermization of automaton
+ * It transforms an automaton with a bitvector formula used a symbol to minterminized version of the automaton.
+ *
+ * Copyright (c) 2022 Martin Hruska <hruskamartin25@gmail.com>
+ *
+ * This file is a part of libmata.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _MATA_MINTERM_HH
+#define _MATA_MINTERM_HH
+
+#include <cudd/cplusplus/cuddObj.hh>
+
+#include <mata/inter-aut.hh>
+
+namespace Mata
+{
+
+class Mintermization
+{
+private: // data types
+    struct BddOrNothing
+    {
+        enum TYPE {NOTHING_E, BDD_E};
+
+        TYPE type;
+        BDD val;
+
+        explicit BddOrNothing(TYPE t) : type(t) {}
+        explicit BddOrNothing(const BDD& bdd) : type(BDD_E), val(bdd) {}
+        BddOrNothing(TYPE t, const BDD& bdd) : type(t), val(bdd) {}
+
+        BddOrNothing operator*(const BddOrNothing& b) const
+        {
+            if (this->type == NOTHING_E)
+                return b;
+            else if (b.type == NOTHING_E)
+                return *this;
+            else
+                return BddOrNothing{BDD_E, this->val * b.val};
+        }
+
+        BddOrNothing operator+(const BddOrNothing& b) const
+        {
+            if (this->type == NOTHING_E)
+                return b;
+            else if (b.type == NOTHING_E)
+                return *this;
+            else
+                return BddOrNothing{BDD_E, this->val + b.val};
+        }
+
+        BddOrNothing operator!() const
+        {
+            if (this->type == NOTHING_E)
+                return BddOrNothing(NOTHING_E);
+            else
+                return BddOrNothing{BDD_E, !this->val};
+        }
+    };
+
+    using DisjunctStatesPair = std::pair<const FormulaGraph *, const FormulaGraph *>;
+
+private: // private data members
+    Cudd bdd_mng;
```

Comment:
I understood right away it is "manager", since I'm familiar with usage of "managers", however, since some younger ones have problem, maybe consider changing it to `mngr`?

Reply:
I understand that it is `manager` as well, but I would like to have a comment such as ` /// Manager for BDDs from library XYZ, handles the creation of BDDs with our input I with these and these characteristics`. This is what I would like to see when looking through the code of Mata itself, without the need to study each and every external dependency. This comment would give the reader the gist of what they can expect. Henceforth, they do not need to study the external library unless they intend to change this specific class and operations with the manager at all. I consider this a better approach to “you have to study everything in order to be able to, for example, change one function in the code.”

Reply:
I added a brief description.